### PR TITLE
Fix accumulo-1244 Update MetadataTableUtil to use the ample api

### PR DESCRIPTION
**Pull request by hkeebler**
Updated methods listed in #1243

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java [hunk 1]
```diff
@@ -489,9 +486,10 @@ public static void deleteTable(TableId tableId, boolean insertDeletes, ServerCon
 
   public static void removeUnusedWALEntries(ServerContext context, KeyExtent extent,
       final List<LogEntry> entries, ZooLock zooLock) {
-    Ample.TabletMutator tabletMutator = context.getAmple().mutateTablet(extent);
-    entries.forEach(tabletMutator::deleteWal);
-    tabletMutator.putZooLock(zooLock);
+    TabletMutator tablet = context.getAmple().mutateTablet(extent);
+    entries.forEach(tablet::deleteWal);
+    tablet.putZooLock(zooLock);
+    tablet.mutate();
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java]: Nice catch.  Seems like this was a bug in the existing code.

**Comment by hkeebler** [comment]
@keith-turner I am trying to understand the ample.readTablet(..) and what is going on in some of the code.  Can you explain some of the following code from the getFileAndLogEntries method?
` public static Pair<List<LogEntry>,SortedMap<FileRef,DataFileValue>>
      getFileAndLogEntries(ServerContext context, KeyExtent extent)
          throws KeeperException, InterruptedException, IOException {
    ArrayList<LogEntry> result = new ArrayList<>();
    TreeMap<FileRef,DataFileValue> sizes = new TreeMap<>();
    VolumeManager fs = context.getVolumeManager();
    TabletMetadata tablet = context.getAmple().readTablet(extent, FILES, LOGS, PREV_ROW, DIR);`
what is PREV_ROW used for here?
` if (!tablet.getExtent().equals(extent))
      throw new RuntimeException("Unexpected extent " + tablet.getExtent() + " expected " + extent);`
why is this checked? what else could it be?

**Comment by keith-turner** [comment]
> why is this checked? what else could it be?

Tablet information is stored in the metadata table using a row of `<table id>;<table end row>`.  A tablets previous end row is stored as a column in the metadata table with the row previously mentioned.

The following call only uses `extent.getEndRow()` to lookup the metadata information.  It also ask it to read the PREV_ROW column.  However, no checks are done by Ampl to ensure that `extent.getPrevEndRow()` matches the value of the PREV_ROW column.  

```java
context.getAmple().readTablet(extent, FILES, LOGS, PREV_ROW, DIR)
```

**Comment by hkeebler** [comment]
Can the following code replacement be done or does "endKey .followingKey(partialKey.ROWCOLFAM)" change the Range of the scan?
```
 public static SortedMap<FileRef,DataFileValue> getDataFileSizes(KeyExtent extent,
    ServerContext context) {
    TreeMap<FileRef,DataFileValue> sizes = new TreeMap<>();
    TabletMetadata tablet = context.getAmple().readTablet(extent, FILES);
    tablet.getFilesMap().forEach((k, v) -> {
      sizes.put(new FileRef(k, context.getVolumeManager().getFullPath(tablet.getTableId(), k)), v);});
    return sizes;
 /* above replaces the following:
   try (Scanner mdScanner = new ScannerImpl(context, MetadataTable.ID, Authorizations.EMPTY)) {
      mdScanner.fetchColumnFamily(DataFileColumnFamily.NAME);
      Text row = extent.getMetadataEntry();
      Key endKey = new Key(row, DataFileColumnFamily.NAME, new Text(""));
      endKey = endKey.followingKey(PartialKey.ROW_COLFAM);
      mdScanner.setRange(new Range(new Key(row), endKey));
      for (Entry<Key,Value> entry : mdScanner) {
        if (!entry.getKey().getRow().equals(row))
          break;
        DataFileValue dfv = new DataFileValue(entry.getValue().get());
        sizes.put(new FileRef(context.getVolumeManager(), entry.getKey()), dfv);
      }
      return sizes;
    }*/
```

**Comment by keith-turner** [comment]
Your replacement looks correct.  Seems like it should get the same information.

`endKey.followingKey(PartialKey.ROW_COLFAM)` creates a key that is the next possible column family after `DataFileColumnFamily.NAME` within the same row.  So I think it is creating a range for a subset of a single row, basically from the empty family to the family after `file`.  If this is right, then the if statement in the loop that checks if the rows are equal should not be needed.  However the fact that the if statement is there makes me wonder if I am missing something.

File: server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java (hunk 1 above)
Comment (ctubbsii) [reply server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java]: This fixed #1217 :+1: :smiley_cat: :tada:

**Pull request merged**